scripts/build.js

```diff
@@ -1,8 +1,73 @@
 const esbuild = require("esbuild");
+const fs = require("node:fs");
+const path = require("node:path");
 
 const LOG_LEVEL = "info";
 const MODULE_BUNDLE_BLOCK_LIST = ["esbuild", "vscode", "./xhr-sync-worker.js"];
 
+// esbuild plugin to Transform VSCode string identifiers like
+// configuration or command IDs from the `continue.` namespace
+// to the `granite.` namespace to avoid clashes
+
+// Quick-and-not-quite-right replacement for not-yet-common RegExp.escape()
+function escapeForRegExp(str) {
+  return str.replace(/([.*+?|^$()[\]{}\\])/g, "\\$1");
+}
+
+const extensionPath = path.resolve(
+  __dirname,
+  "../continue/extensions/vscode/src",
+);
+const extensionFilter = new RegExp(
+  "^" + escapeForRegExp(extensionPath) + "/.*\\.ts$",
+);
+
+const namespaceSubstitutions = [
+  {
+    pattern: /^continue[.]/,
+    replacement: "granite.",
+  },
+  {
+    pattern: /^Continue[.]continue$/,
+    replacement: "redhat.granite-code",
+  },
+  {
+    pattern: /continueGUI/,
+    replacement: "graniteGUI",
+  },
+];
+
+function namespaceTransform(fileContents) {
+  // For speed, we do this just as a string replacement, rather than
+  // using ts-morph or similar. We only care about simple strings
+  // like "continue.acceptDiff", so we can simplify matching
+  // strings in the source code.
+  return fileContents.replace(/"([^"\\\r\n]+)"/g, (m, strContents) => {
+    for (const { pattern, replacement } of namespaceSubstitutions) {
+      strContents = strContents.replace(pattern, replacement);
+    }
+    return '"' + strContents + '"';
+  });
+}
+
+let namespaceTransformPlugin = {
+  name: "example",
```

Comment:
should use a better name here... maybe `granite-namespace-transform` or something like that